### ISSUE TITLE
Reaction dependencies

### ISSUE THREAD

**Issue by cthoyt** [pybel/bel2abm#1]
Dependency of a reaction and all kinds of relationships needs to become also dependent on the concentration of agents in the reaction distance. So far, a random choice is made whenever an agent can participate in more than 1 reaction/relationship (eg, whenever it can participate in n actions, the action to be performed is chosen by “random n”). Only the one chosen is then performed, without looking at concentrations.

**Comment by michaelaguendel** [comment]
True, you can add it to the future work if you like. Random n doesn't
depend on the concentratiin, but, in case random n will execute the
reaction, the higher the concentration of a substance, the higher the
possibility the relative substance will be chosen. But you are right this
is only an indirect probability increase of a highly concentrated substance
to participate in the reaction.

For the rest, how is it going?

Il 9 dic 2017 19:22, "Charles Tapley Hoyt" <notifications@github.com> ha
scritto:

**Comment by michaelaguendel** [comment]
Ps sorry only saw your message today

Il 9 dic 2017 19:22, "Charles Tapley Hoyt" <notifications@github.com> ha
scritto:
